Client/Frontend/Browser/URLBarView.swift

```diff
@@ -747,7 +747,7 @@ extension URLBarView: AppStateDelegate {
         if toolbarIsShowing {
             let showShareButton = HomePageAccessors.isButtonInMenu(appState)
             homePageButton.hidden = showShareButton
-            shareButton.hidden = !showShareButton
+            shareButton.hidden = !showShareButton || inOverlayMode
```

Comment:
I did initially try that, but the problem is then it will show the home button because the .hidden of homePageButton is the inverse of the .hidden for shareButton

![screen shot 2016-06-29 at 1 36 27 pm](https://cloud.githubusercontent.com/assets/11801330/16462328/b8b1630e-3dfe-11e6-8967-d57d9de6ad36.png)